### ISSUE TITLE
Small patch to match a specific string in body(or header) for healthness

### ISSUE THREAD

**Issue by ut0mt8** [openresty/lua-resty-upstream-healthcheck#29]
Useful in some situation, you want to check something more precise than an http status.

``
--- healthcheck.lua.orig	2016-11-19 18:07:12.494050376 +0000
+++ healthcheck.lua	2016-11-19 18:09:15.370479478 +0000
@@ -6,6 +6,7 @@
 local DEBUG = ngx.DEBUG
 local str_find = string.find
 local sub = string.sub
+local re_match = ngx.re.match
 local re_find = ngx.re.find
 local new_timer = ngx.timer.at
 local shared = ngx.shared
@@ -214,6 +215,7 @@
     local name = peer.name
     local statuses = ctx.statuses
     local req = ctx.http_req
+    local match_string = ctx.match_string

     local sock, err = stream_sock()
     if not sock then
@@ -273,6 +275,18 @@
         end
     end

+    if match_string then
+        local data, err = sock:receive('*a')
+        local m, err = re_match(data, match_string, "jo")
+        if not m then
+            peer_error(ctx, is_backup, id, peer,
+                       "match_string not found from ", name, ": ",
+                       data)
+            sock:close()
+            return
+        end
+    end
+
     peer_ok(ctx, is_backup, id, peer)
     sock:close()
 end
@@ -567,6 +581,11 @@

     -- debug("interval: ", interval)

+    local match_string = opts.match_string
+    if not match_string then
+        match_string = "OKI"
+    end
+
     local concur = opts.concurrency
     if not concur then
         concur = 1
@@ -617,6 +636,7 @@
         dict = dict,
         fall = fall,
         rise = rise,
+        match_string = match_string,
         statuses = statuses,
         version = 0,
         concurrency = concur,

**Comment by agentzh** [comment]
@ut0mt8 It's recommended to create github pull request to simplify code review and patch merging on our side. Thank you.

**Comment by agentzh** [comment]
@ut0mt8 BTW, `receive("*a")` may be VERY expensive when the response is huge. Don't do that.

**Comment by ut0mt8** [comment]
Well I see there is already a similar pull request, I think my version is more simple, I can just switch to receive(2048) or whatever reasonable value. Anyway a page which a healtcheck must be simple, Is not a good idea to check the regular homepage :)

**Comment by wtiancai** [comment]
i agree ut0mt8 opnion，the test page may has very simple content，just like ok
